engine/components/sql/src/main/scala/pl/touk/nussknacker/sql/db/query/QueryExecutor.scala

```diff
@@ -0,0 +1,51 @@
+package pl.touk.nussknacker.sql.db.query
+
+import pl.touk.nussknacker.engine.api.typed.TypedMap
+import pl.touk.nussknacker.sql.db.schema.TableDefinition
+
+import java.sql.{PreparedStatement, ResultSet}
+import scala.collection.mutable.ArrayBuffer
+
+trait QueryExecutor {
+
+  type QueryResult
+
+  def execute(statement: PreparedStatement): QueryResult
+
+  protected def toTypedMap(tableDef: TableDefinition, resultSet: ResultSet): TypedMap = {
+    val fields = tableDef.columnDefs.map { columnDef =>
+      columnDef.name -> resultSet.getObject(columnDef.no)
+    }.toMap
+    TypedMap(fields)
+  }
+}
+
+
+class SingleResultQueryExecutor(tableDef: TableDefinition) extends QueryExecutor {
+
+  override type QueryResult = TypedMap
+
+  def execute(statement: PreparedStatement): QueryResult = {
+    val resultSet = statement.executeQuery()
+    if (resultSet.next())
+      toTypedMap(tableDef, resultSet)
+    else
+      null
+  }
+}
+
+class ResultSetQueryExecutor(tableDef: TableDefinition) extends QueryExecutor {
+
+  import scala.collection.JavaConverters.bufferAsJavaListConverter
+
+  override type QueryResult = java.util.List[TypedMap]
+
+  override def execute(statement: PreparedStatement): QueryResult = {
+    val resultSet = statement.executeQuery()
+    val results = new ArrayBuffer[TypedMap]()
```

Comment:
let's just use ArrayList to avoid `.asJava`

Reply:
done